### PR TITLE
Pull back MSRV on Windows

### PR DESCRIPTION
**Pull request by workingjubilee**

### CODE REVIEW

File: .github/workflows/check-binary-size.yml [hunk 1]
```diff
@@ -6,8 +6,9 @@ name: Check binary size
 
 on:
   pull_request_target:
-    branches:
-      - master
+    # HACK(jubilee): something broke the distributed LLVM libso and I don't know what.
```
Comment (ChrisDenton) [inline .github/workflows/check-binary-size.yml:9]: this hack not work?

Comment (Nilstrieb) [inline .github/workflows/check-binary-size.yml:9]: unsure what this is referring to, but the LLVM libso got renamed into a linker script with a normal name the fully versioned name pointed to by it

Comment (ChrisDenton) [reply .github/workflows/check-binary-size.yml:9]:
```
build/host/stage0-sysroot/bin/rustc: error while loading shared libraries: libLLVM.so.18.1-rust-1.78.0-beta: cannot open shared object file: No such file or directory
```

Comment (workingjubilee) [reply .github/workflows/check-binary-size.yml:9]: needs to be on master for it to matter, afaik.

**Review by ChrisDenton — APPROVED** [review]
Ok, let's do this! A bunch of hacks is better than not having CI working.

File: .github/workflows/check-binary-size.yml (hunk 1 above)
Comment (Nilstrieb) [reply .github/workflows/check-binary-size.yml:9]: huh, wtf

Comment (Nilstrieb) [reply .github/workflows/check-binary-size.yml:9]: https://github.com/rust-lang/rust/issues/122913

**Pull request merged**